cluster-autoscaler/core/scaleup/resource/manager_test.go

```diff
@@ -170,6 +171,63 @@ func TestApplyLimits(t *testing.T) {
 	}
 }
 
+type updateLimitsTestCase struct {
+	resourcesLeft  Limits
+	resourcesDelta Delta
+	nodeCount      int
+	expectedLeft   Limits
+	expectedErr    error
+}
+
+func TestUpdateLimits(t *testing.T) {
+	testCases := []updateLimitsTestCase{
+		{
+			resourcesLeft:  Limits{"cpu": 10, "memory": 20},
+			resourcesDelta: Delta{"cpu": 8, "memory": 16},
+			nodeCount:      1,
+			expectedLeft:   Limits{"cpu": 2, "memory": 4},
+			expectedErr:    nil,
+		},
+		{
+			resourcesLeft:  Limits{"cpu": 10, "memory": 20},
+			resourcesDelta: Delta{"cpu": 8, "memory": 16},
+			nodeCount:      2,
+			expectedLeft:   Limits{"cpu": 10, "memory": 20},
+			expectedErr:    fmt.Errorf("cannot create 2 nodes: cpu resource left (10) is less than resource required (16)"),
```

Comment:
I think this test case indeterministic. The Limits are just maps, and in `golang` maps are not ordered. Because of that it's possible that the error here would be the memory related one. I think the cpu one is more likely based on the map internal implementation, but in ~1/8 of runs the memory limit will be checked first. You can check it by running this using `go test` with `-count 1000` flag - it should fail fairly quickly.

Reply:
Good catch! Just updated this test case to a deterministic one, thanks.